beesdoo_shift/wizard/subscribe.py

```diff
@@ -207,4 +198,13 @@ def subscribe(self):
             # to a shift to keep information like "Worker mode, session info
             # ,...
             status_id.sudo().write(data)
+
+        if self.shift_id:
+            # Remove existing shift then subscribe to the new shift
+            self.cooperator_id.sudo().write(
+                {"subscribed_shift_ids": [(6, 0, [self.shift_id.id])]}
+            )
+        if self.working_mode != "regular":
+            # Remove existing shift then subscribe to the new shift
+            self.cooperator_id.sudo().write({"subscribed_shift_ids": [(5,)]})
```

Comment:
@vvrossem Is this to cover the case when an irregular worker becomes a regular worker ? If so, this test should remain before the write / create.

Reply:
no, it's to fix a newly bug : error message "working mode is not correctly defined" when trying to subscribe a member to a shift and defining first the working mode

steps to reproduce the bug were :

- go on a cooperator form (Contacts)

- no working mode, no shift assigned

- click on button Subscribe to shift

- choose : Working mode = Regular worker, Choose Shift = X_Day_Time_Item

- Confirm => Error is raised: Working mode is not properly defined. Please check if the worker is subscribed